src/Bridge/Elasticsearch/DataProvider/CollectionDataProvider.php

```diff
@@ -38,20 +38,20 @@ final class CollectionDataProvider implements ContextAwareCollectionDataProvider
     private $client;
     private $documentMetadataFactory;
     private $identifierExtractor;
-    private $denormalizer;
+    private $paginatorFactory;
     private $pagination;
     private $resourceMetadataFactory;
     private $collectionExtensions;
 
     /**
      * @param RequestBodySearchCollectionExtensionInterface[] $collectionExtensions
      */
-    public function __construct(Client $client, DocumentMetadataFactoryInterface $documentMetadataFactory, IdentifierExtractorInterface $identifierExtractor, DenormalizerInterface $denormalizer, Pagination $pagination, ResourceMetadataFactoryInterface $resourceMetadataFactory, iterable $collectionExtensions = [])
```

Comment:
It's a BC break. Can we avoid it by making the factory optional and just create the old service if not passed?

Reply:
The class is already final so I don't understand why it could be a BC break

Reply:
Requested changes have been made ;)

Reply:
The new argument must be the last one to prevent the break.

Reply:
The way I implemented the fix should be BC. I've simply remove the typehint on the `Denormalizer` argument to allow passing either a denormalizer or a paginator factory instance. When a denormalizer is given (aka legacy behavior), a deprecation warning message is raised to inform the user that passing a denormalizer will no longer be supported in APIP 3 in favor of passing a `PaginatorFactory` instance instead. When a denormalizer is given, it's wrapped into a paginator factory. So the code should be BC.

Anyway, I made another commit introducing a last optional argument for the paginator factory dependency.